pkg/tests/accounts_test.go

```diff
@@ -452,6 +457,219 @@ func Test_ImplicitAccounts(t *testing.T) {
+
+		validatorBIC -= burned
+		walletBIC -= burned
+
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              node1.Validator.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(validatorBIC, block1Slot),
+			OutputID:        validatorAccountOutput.OutputID(),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			BlockIssuerKeys: node1.Validator.BlockIssuerKeys(),
+			StakeEndEpoch:   iotago.MaxEpochIndex,
+			ValidatorStake:  mock.MinValidatorAccountAmount,
+		}, ts.Nodes()...)
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              wallet.BlockIssuer.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(walletBIC, block1Slot),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			OutputID:        blockIssuerAccountOutput.OutputID(),
+			BlockIssuerKeys: wallet.BlockIssuer.BlockIssuerKeys(),
+		}, ts.Nodes()...)
+	}
+
+	block2Slot := latestParent.ID().Slot()
+
+	// Try to issue more blocks from each of the issuers - one succeeds in issuing a block,
+	// the other has the block rejected in the CommitmentFilter as his account has negative BIC value.
+	{
+
+		block21 := ts.IssueBasicBlockAtSlotWithOptions("block2.1", block2Slot, ts.Wallet("node1"), &iotago.TaggedData{})
+
+		block22 := ts.IssueBasicBlockAtSlotWithOptions("block2.2", block2Slot, wallet, &iotago.TaggedData{}, mock.WithStrongParents(ts.BlockID("block2.1")))
+
+		ts.AssertBlockFiltered([]*blocks.Block{block22}, iotago.ErrNegativeBIC, wallet.Node)
+
+		latestParent = ts.CommitUntilSlot(ts.BlockID("block2.1").Slot(), block21)
+
+		burned := iotago.BlockIssuanceCredits(block21.WorkScore()) * iotago.BlockIssuanceCredits(ts.API.ProtocolParameters().CongestionControlParameters().MinReferenceManaCost)
+
+		validatorBIC -= burned
+
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              node1.Validator.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(validatorBIC, block2Slot),
+			OutputID:        validatorAccountOutput.OutputID(),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			BlockIssuerKeys: node1.Validator.BlockIssuerKeys(),
+			StakeEndEpoch:   iotago.MaxEpochIndex,
+			ValidatorStake:  mock.MinValidatorAccountAmount,
+			LatestSupportedProtocolVersionAndHash: model.VersionAndHash{
+				Version: ts.API.ProtocolParameters().Version(),
+				Hash:    lo.PanicOnErr(ts.API.ProtocolParameters().Hash()),
+			},
+		}, ts.Nodes()...)
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              wallet.BlockIssuer.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(walletBIC, block1Slot),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			OutputID:        blockIssuerAccountOutput.OutputID(),
+			BlockIssuerKeys: wallet.BlockIssuer.BlockIssuerKeys(),
+		}, ts.Nodes()...)
+	}
+
+	block3Slot := latestParent.ID().Slot()
+
+	// Allot some mana to the locked account to unlock it.
+	{
+		allottedBIC := iotago.BlockIssuanceCredits(500)
+		tx1, err := wallet.CreateSignedTransactionWithOptions(
```

Comment:
create a wallet method called AllotManaToWallet() rather than using the createSignedTransactionWithOptions() method.

Reply:
Add a comment here to to highlight that the locked wallet is creating the transaction, but not issuing the block.